### PR TITLE
Auto split insert data into multiple batches.

### PR DESCRIPTION
**Pull request by shenli**
Introduce a new session scope system variable `tidb_batch_insert `. If set it to `on` or 1, we will automatically split large insert data into small batches.

I will try to add more tests in the next commit.

### CODE REVIEW

**Comment by shenli** [comment]
@coocood @zimulala @tiancaiamao PTAL

File: executor/write.go [hunk 1]
```diff
@@ -626,14 +630,25 @@ func (e *InsertExec) Next() (*Row, error) {
 		return nil, errors.Trace(err)
 	}
 
+	rowCount := 0
+	batchInsert := e.ctx.GetSessionVars().BatchInsert
```
Comment (hanfei1991) [inline executor/write.go]: Do we need to check it is auto-commit ?

**Comment by shenli** [comment]
@hanfei1991 PTAL

**Comment by hanfei1991** [comment]
LGTM

**Comment by zimulala** [comment]
If this `insert statement` split into two batches, the first transaction commits successfully, the second transaction is failed, and the failure that can be retried. Then the retry results may be problematic.

**Comment by shenli** [comment]
@zimulala PTAL

**Comment by shenli** [comment]
@zimulala PTAL

**Pull request merged**